docs/source/Mapping.rst

```diff
@@ -0,0 +1,175 @@
+
+
+Now assume this circuit shall be mapped to a 4-qubit architecture defined by the following coupling map:
+
+.. image:: /images/linear_arch.svg
+   :width: 80%
+   :alt: 5-qubit Architecture
+   :align: center
+
+
+In *QMAP* this architecture can be manually defined.
+
+    .. code-block:: python3
+
+       arch = qmap.Architecture(
+           4,
+           {
+               (0, 1),
+               (1, 0),
+               (1, 2),
+               (2, 1),
+               (2, 3),
+               (3, 2),
+           },
+       )
+
+The quantum circuit :code:`qc` can not be run directly on this architecture since it contains gates that act on qubits not connected on the device architecture.
+Naively inserting SWAP gates that permute the logical-to-physical qubit mapping on the fly may yield the following compiled circuit.
+
+    .. code-block:: console
+
+                      ┌───┐                      ░ ┌───┐ ░                       ░ ┌─┐
+          q_0 -> q_0: ┤ H ├──■───X───────────────░─┤ T ├─░───────────────X───■───░─┤M├─────────
+                      └───┘┌─┴─┐ │               ░ ├───┤ ░               │ ┌─┴─┐ ░ └╥┘┌─┐
+          q_1 -> q_1: ─────┤ X ├─X───■───X───────░─┤ T ├─░───────X───■───X─┤ X ├─░──╫─┤M├──────
+                           └───┘   ┌─┴─┐ │       ░ ├───┤ ░       │ ┌─┴─┐   └───┘ ░  ║ └╥┘┌─┐
+          q_2 -> q_2: ─────────────┤ X ├─X───■───░─┤ T ├─░───■───X─┤ X ├─────────░──╫──╫─┤M├───
+                                   └───┘   ┌─┴─┐ ░ ├───┤ ░ ┌─┴─┐   └───┘         ░  ║  ║ └╥┘┌─┐
+          q_3 -> q_3: ─────────────────────┤ X ├─░─┤ T ├─░─┤ X ├─────────────────░──╫──╫──╫─┤M├
+                                           └───┘ ░ └───┘ ░ └───┘                 ░  ║  ║  ║ └╥┘
+              meas: 4/══════════════════════════════════════════════════════════════╩══╩══╩══╩═
+                                                                                    0  1  2  3
+
+Over the course of the mapping, *four* SWAP gates have been introduce to satisfy the connectivity constraints of the device's architecture.
+Since every additional gate increases the probability of errors, this is a very costly overhead for such a small circuit.
+
+Keeping the number of additionally introduced gates as small as possible is key for ensuring the successful execution of the quantum circuit. Finding an optimal mapping for a quantum circuit is an NP-hard problem :cite:labelpar:`boteaComplexityQuantumCircuit2018`.
+*QMAP* offers two dedicated techniques for tackling that problem:
+- An *exact* mapping approach (based on :cite:labelpar:`willeMappingQuantumCircuits2019, burgholzer2022limitingSearchSpace`) that guarantees (gate-optimal) solutions and is typically suitable for up to 8 qubits.
+- A *heuristic* mapping approach (based on :cite:labelpar:`zulehnerEfficientMethodologyMapping2019, hillmichExlpoitingQuantumTeleportation2021`) that allows to determine efficient mapping solutions in a scalable fashion for up to hundreds of qubits.
```

Comment:
for some reason the link previews for the second citation in `[3, 4]` and `[5, 6]`do not work properly in the docs. maybe split these into separate citations, i.e., `[3], [4]`.

Reply:
Done